image/build.sh

```diff
@@ -276,21 +191,34 @@ function install_openssl()
 		# shellcheck disable=SC2030,SC2001
 		CFLAGS=$(adjust_optimization_level "$STATICLIB_CFLAGS")
 		export CFLAGS
+		# shellcheck disable=SC2153
+		export LIB_CFLAGS="$SHLIB_CFLAGS"
+		# shellcheck disable=SC2153
+		export LIB_LDFLAGS="-Wl,-znodelete -shared -Wl,-Bsymbolic $SHLIB_LDFLAGS"
+		export DSO_CFLAGS="$SHLIB_CFLAGS"
+		export DSO_LDFLAGS="-Wl,-z,defs -Wl,-znodelete -shared -Wl,-Bsymbolic $SHLIB_LDFLAGS"
 
 		# shellcheck disable=SC2086
 		run ./Configure "linux-$(uname -m)" \
-		    --prefix="$PREFIX" --openssldir="$PREFIX/openssl" \
-		    threads zlib no-shared no-sse2 -fvisibility=hidden $CFLAGS $LDFLAGS
-		run make
+			--prefix="$PREFIX" --openssldir="$PREFIX/openssl" \
+			threads zlib no-shared no-sse2 no-legacy no-tests
```

Comment:
Maybe OpenSSL should be built with `no-module` to be fully static (i.e. have no dynamic providers). At least that's a change I did on my fork trying to fix the image.